hlf/scala/admission.md

```diff
@@ -0,0 +1,73 @@
+# Hyperledger Scala API Admission
+
+In the following, the hlf-api for the "admission" handling will be explained.
+For insight in the General hlf-api, please refer to it's .Readme - File.
+> [!NOTE]
+Any Method can throw these [Exceptions](errors.md#Exceptions).
+
+For "admission" handling we offer the following interface. 
+Additionally, as described in [General Communication](general-communication.md), different proposal calls can be made.
+
+
+## AddAdmission(enrollmentId: String, courseId: String, moduleId: String, timestamp: Option[String])
+- Returns
+    - Admission :: Json (refer to: [Admission](../chaincode/admission.md#Admission))
+        - => Success
+    - TransactionError :: Json (refer to: [DetailedError](../chaincode/errors.md#DetailedError))
+        - => error is returned
+          - If the given parameters could not be parsed.
+    - TransactionError :: Json (refer to: [SemanticError](../chaincode/errors.md#SemanticError))
+        - => error is returned
+          - If the given transaction could not be performed. ValidationRules were violated.
+
+## DropAdmission(admissionId: String)
+- Returns
+    - "": String
+        - => Success
+    - TransactionError :: Json (refer to: [GenericError](../chaincode/errors.md#GenericError))
+        - => error is returned, 
+          - If the admissionId specified is not present on the ledger.
+          - If the required approvals are not present on the ledger.
+    - TransactionError :: Json (refer to: [DetailedError](../chaincode/errors.md#DetailedError))
+        - => error is returned
+          - If the given parameters could not be parsed.
+
+## GetAdmissionsForUser(enrollmentId: String)

```

Comment:
For the frontend we offer one endpoint in which query parameters determine the filters. We also defined that if no parameters are given, we return ALL admissions (mostly for our admin debug purposes).

So we either need another method to fetch all admissions, or we merge all these methods into one.
I prefer merging, since multiple query parameters set by the frontend mean that we either invoke multiple methods, or just fetch all and filter, both suboptimal.

Reply:
Did you write some DSL for those queries? How is the chaincode supposed to know what to filter for?
Adding GetAll is a good idea though !

Reply:
On our side these parameters are registered as Options, being None if they are not set. It would result in a lot less work on our side if the api supported using the same parameters as Options. So something like
```
GetAdmissions(username: Option[String], courseId: Options[String], moduleId: Option[String])
```
If all options are None, it should return ALL admissions.

Reply:
looks promising, so we are not using general queries but designated named/indexed parameters for the filters.
Will change this tomorrow and re-request reviews

Reply:
filter change was applied